### PR TITLE
BuildAPIKey: remove old revoked/expired keys

### PR DESCRIPTION
**Pull request by stsewd**
Closes https://github.com/readthedocs/readthedocs.org/issues/10764

### CODE REVIEW

File: readthedocs/api/v2/tasks.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+log = structlog.get_logger(__name__)
+
+
+@app.task(queue="web")
+def delete_old_revoked_build_api_keys():
+    """
+    Delete revoked and expired keys that are older than 7 days.
+
+    We don't delete keys created in the last 7 days,
+    to have some audit trail in case we need to investigate something.
```
Comment (stsewd) [inline readthedocs/api/v2/tasks.py:16]: Can be 15 days too if we want.

Comment (ericholscher) [reply readthedocs/api/v2/tasks.py:18]: Seems fine to start with 15. 👍

File: readthedocs/api/v2/tasks.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    """
+    created_before = timezone.now() - timedelta(days=7)
+    to_delete = BuildAPIKey.objects.filter(revoked=True, created__lt=created_before)
```
Comment (ericholscher) [inline readthedocs/api/v2/tasks.py:21]: I know in the past these queries have been too slow and gotten cancelled, and we've moved to raw SQL. I guess we can try this and migrate if it's too slow?

**Review by ericholscher — APPROVED** [review]

**Comment by stsewd** [comment]
Looks like it did time out... I'm manually deleting the old keys from web-extra for now, hopefully the task doesn't time out tomorrow.

**Pull request merged**